### PR TITLE
Server cleanup and refactor

### PR DESCRIPTION
**Pull request by bryevdv**
Some of this is covered by #2029

### CODE REVIEW

**Comment by bryevdv** [comment]
ping @mattpap @damianavila @fpliger @havocp The PR has been re-opened here, against `dev_0.10`

**Comment by havocp** [comment]
Looks really good!

**Comment by bryevdv** [comment]
@bokeh/dev last chance for comments, I plan to merge this in an hour or so. The 0.10 release is a ways off so there is plenty of time for more refinements (I already have more branches on the way in fact)

**Comment by fpliger** [comment]
LGTM. Nice improvements.

File: bokeh/resources.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ class Resources(object):
     _default_root_dir = "."
     _default_root_url = "http://127.0.0.1:5006/"
 
-    logo_url = "http://bokeh.pydata.org/_static/bokeh-transparent.png"
+    logo_url = "http://bokeh.pydata.org/static/bokeh-transparent.png"
```
Comment (damianavila) [inline bokeh/resources.py]: this os really the correct url? I don't see anything hitting on any of those...

File: setup.py [hunk 2]
```diff
@@ -546,11 +546,14 @@ def parse_jsargs():
...
-        'bokeh.server.views',
         'bokeh.server.blaze',
-        'bokeh.server.utils',
+        'bokeh.server.models',
+        'bokeh.server.storage',
         'bokeh.server.tests',
+        'bokeh.server.utils',
+        'bokeh.server.views',
+        'bokeh.server.websocket',
+        'bokeh.server.zmq',
```
Comment (damianavila) [inline setup.py]: In other PR, we should auto-detect all of these...

**Comment by damianavila** [comment]
Just minor comments... otherwise, LGTM...

File: setup.py (hunk 2 above)
Comment (bryevdv) [inline setup.py]: Maybe, there are arguments for and against.

File: bokeh/resources.py (hunk 1 above)
Comment (bryevdv) [inline bokeh/resources.py]: I updated the web server with the image.

**Comment by bryevdv** [comment]
ok, will merge as soon as travis passes

**Comment by damianavila** [comment]
@bryevdv this is failing well because of some import error...

**Pull request merged**